test/lightning_web/plugs/webhook_auth_test.exs

```diff
@@ -7,15 +7,89 @@ defmodule LightningWeb.Plugs.WebhookAuthTest do
   alias LightningWeb.Plugs.WebhookAuth
   alias Lightning.Repo
 
+  @moduletag capture_log: true
+
   setup do
     {:ok, trigger: insert(:trigger), auth_method: insert(:webhook_auth_method)}
   end
 
+  test "OPTIONS preflight is a no-op" do
+    conn = conn(:options, "/i/whatever") |> WebhookAuth.call([])
+    refute conn.halted
+    assert conn.status == nil
+    assert conn.assigns[:trigger] == nil
+  end
+
+  test "non-/i path passes through unchanged" do
+    conn = conn(:post, "/api/things") |> WebhookAuth.call([])
+    refute conn.halted
+    assert conn.status == nil
+    assert conn.assigns[:trigger] == nil
+  end
+
+  test "responds 404 for wrong x-api-key on protected trigger", %{
+    trigger: trigger
+  } do
+    api_method = insert(:webhook_auth_method, auth_type: :api, api_key: "secret")
+    associate_auth_method(trigger, api_method)
+
+    conn =
+      conn(:post, "/i/#{trigger.id}")
+      |> put_req_header("x-api-key", "nope")
+      |> WebhookAuth.call([])
+
+    assert conn.halted
+    assert conn.status == 404
+    assert Jason.decode!(conn.resp_body) == %{"error" => "webhook_not_found"}
+  end
+
+  test "assigns trigger for matching x-api-key", %{trigger: trigger} do
+    api_method = insert(:webhook_auth_method, auth_type: :api, api_key: "secret")
+    associate_auth_method(trigger, api_method)
+
+    conn =
+      conn(:post, "/i/#{trigger.id}")
+      |> put_req_header("x-api-key", "secret")
+      |> WebhookAuth.call([])
+
+    expected_trigger =
+      trigger
+      |> unload_relation(:workflow)
+      |> Repo.preload([:workflow, :edges, :webhook_auth_methods])
+
+    refute conn.halted
+    assert conn.assigns[:trigger] == expected_trigger
+  end
+
+  test "returns 503 with Retry-After when DB lookup errors are exhausted" do
+    Mimic.copy(Lightning.Retry)
+
+    Mimic.expect(Lightning.Retry, :with_webhook_retry, fn _fun, _opts ->
+      {:error, %DBConnection.ConnectionError{message: "db down"}}
```

Comment:
Does Mimic allow matching of arguments passed to the method being mocked? If not, I imagine you could perhaps do pattern matching in the function that handles the response?

For me, when I am writing a test where the code is calling another function, the two questions I want answered are:

- Does the code I am testing call the method correctly?
- Does the code I am testing do the right thing with the method response?

I think we may be missing coverage of the former, so it would be great if we could use Mimic for that.